packages/gatsby-source-wordpress/src/normalize.js

```diff
@@ -0,0 +1,385 @@
+      const me = media.find(m => m.source_url === e.acf.linked_image)
+      e.acf.linked_image___NODE = me.id
+      delete e.acf.linked_image
+    }
+
+    const hasPhoto = object => _.some(object, value => isPhoto(value))
+
+    const isPhoto = field =>
+      _.isObject(field) &&
+      field.wordpress_id &&
+      field.url &&
+      field.width &&
+      field.height
+        ? true
+        : false
+
+    const replacePhoto = field =>
+      media.find(m => m.wordpress_id === field.wordpress_id).id
+
+    const replaceFieldsInObject = object => {
+      _.each(object, (value, key) => {
+        if (_.isArray(value)) {
+          return value.forEach(v => replaceFieldsInObject(v))
+        }
+        if (isPhoto(value)) {
+          object[`${key}___NODE`] = replacePhoto(value)
+          delete object[key]
+        }
+      })
+    }
+
+    if (e.acf && e.acf.page_builder) {
+      e.acf.page_builder = e.acf.page_builder.map(f => {
+        replaceFieldsInObject(f)
+        return f
+      })
+    }
+    return e
+  })
+}
+
+// Downloads media files and removes "sizes" data as useless in Gatsby context.
+exports.downloadMediaFiles = async ({ entities, store, cache, createNode }) =>
+  Promise.all(
+    entities.map(async e => {
+      let fileNode
+      if (e.__type === `wordpress__wp_media`) {
+        try {
+          fileNode = await createRemoteFileNode({
+            url: e.source_url,
+            store,
+            cache,
+            createNode,
+          })
+        } catch (e) {
+          // Ignore
+        }
+      }
+
+      if (fileNode) {
+        e.localFile___NODE = fileNode.id
+        delete e.media_details.sizes
+      }
+
+      return e
+    })
+  )
+
+const createACFChildNodes = (
+  obj,
+  entityId,
+  topLevelIndex,
+  type,
+  children,
+  createNode
+) => {
+  // Replace any child arrays with pointers to nodes
+  _.each(obj, (value, key) => {
+    if (_.isArray(value)) {
+      obj[`${key}___NODE`] = value.map(v =>
+        createACFChildNodes(
+          v,
+          entityId,
+          topLevelIndex,
+          type + key,
+          children,
+          createNode
+        )
+      )
+      delete obj[key]
+    }
+  })
+  // obj = deepMapKeys(
+  // obj,
+  // key => (key === `ID` ? getValidKey({ key: `id` }) : getValidKey({ key }))
+  // )
+  // console.log(obj)
+  const acfChildNode = {
+    ...obj,
+    id: entityId + topLevelIndex + type,
+    parent: entityId,
+    children: [],
+    internal: { type, contentDigest: digest(JSON.stringify(obj)) },
+  }
+  createNode(acfChildNode)
+  children.push(acfChildNode.id)
+
+  return acfChildNode
+}
+
+exports.createNodesFromEntities = ({ entities, createNode }) => {
+  entities.forEach(e => {
+    // Create subnodes for page_builder
+    // find any "rendered" field and make that top-level
+    let { __type, ...entity } = e
+    let children = []
+    // Create child nodes for acf.page_builder
+    if (entity.acf && entity.acf.page_builder) {
```

Comment:
Can you make this more generic ? The node `page_builder` can receive any name, it does not identify a node of type "ACF Flexible". One way to do this : if the value is an array and its first element has a property named `acf_fc_layout` then it's probably a "ACF Flexible" type component.

The node `page_builder` can also have siblings at the same level, being another node of type "ACF Flexible", or of other types.
![image](https://user-images.githubusercontent.com/5808108/30176272-24540b98-9402-11e7-97af-40dfe661b5e4.png)